src/quo2/components/settings/settings_list/view.cljs

```diff
@@ -0,0 +1,119 @@
+            [quo2.components.markdown.text :as text]
+            [quo2.foundations.colors :as colors]
+            [react-native.core :as rn]))
+
+(defn settings-title
+  [title]
+  [rn/view
+   {:flex 1}
+   [text/text
+    {:accessibility-label :setting-item-name-text
+     :ellipsize-mode      :tail
+     :style               style/title
+     :number-of-lines     1
+     :weight              :medium
+     :size                :paragraph-1}
+    title]
+  ])
+
+(def browser-context-icon
+  [rn/view
+   [quo2.icons/icon :browser-context
+    {:container-style style/icon
+     :color           (colors/theme-colors
+                       colors/neutral-50
+                       colors/neutral-40)}]])
+
+(def chevron-icon
+  [rn/view
+   [quo2.icons/icon :chevron-right
+    {:color (colors/theme-colors
+             colors/neutral-50
+             colors/neutral-40)}]])
+
+(defn switch-button
+  [{:keys [:value
+           on-value-change]}]
+  (js/console.warn on-value-change)
+  [rn/switch
+   {:value                value
+    :on-value-change      (fn [new-value] (on-value-change new-value))
+    ;; TODO: For some reason this prop is not working on ios simulator, spike a little bit more
+    :ios-background-color (colors/theme-colors
+                           colors/neutral-30
+                           colors/neutral-80)
+    ;; TODO: Ask for colors on the track, there is no definition on the design
+    ;; Colors for false value only work on Android
+    ;; :track-color          {:false (colors/theme-colors
+    ;;                               colors/danger
+    ;;                               colors/danger)
+    ;;                       :true  (colors/theme-colors
+    ;;                               colors/danger
+    ;;                               colors/danger)}
+    :style                style/switch-button}]
+)
+
+(def badge-icon
+  [rn/view
+   {:style style/dot}])
+
+(defn right-button
+  [{:keys [title
+           on-press]}]
+  [rn/touchable-opacity
+   {:on-press on-press
+    :style    style/right-button}
+   [text/text
+    {:accessibility-label :settings-right-button-title
+     :number-of-lines     1
+     :weight              :medium
+     :style               style/right-button-title
+     :size                :paragraph-2}
+    title]]
+)
+
+(defn communities-icons
+  [{:keys [data
+           icon-style]}]
+  (let [communities-count (- (count data) 1)]
+    [rn/view
+     {:style {:flex 1 :margin-right 12}}
+     (map-indexed
+      (fn [index {:keys [source accesibility-label]}]
+        [rn/image
+         {:key                source
+          :source             {:uri source}
+          :accesibility-label accesibility-label
+          :style              (merge (style/community-icon (- communities-count index)) icon-style)}])
+      data)]))
+
+(defn settings-list
+  [{:keys [title
+           on-press
+           accessibility-label
+           left-icon
+           chevron
+           switch-props
```

Comment:
I think we can make the component API more extensible. Instead of passing `switch-props`, `button-props`, etc, we can use something like `action` like in the design and let the parent component pass the appropriate component. Maybe @J-Son89 can give more insight on this.

Reply:
Yeah I believe in the past we have had similar discussions to make our components more flexible this way. I think given our use case that it's to a specific design that the designers have set we are better off just having everything configurable and not needing much work to be done by the developers when implementing screens with these components. 
I think we would approach differently if we were making a reusable component library but given our use case this is the better approach. 
Very open to further discussion on this and I think it would be good to include the design team and product for where Status is expected to be going, as perhaps long term the approach you have suggested @ajayesivan is the better solution 👍

Reply:
I think this approach makes more sense for this component since there are only a few options to take care of. As you said my previous suggestion will put more work in the hands of the developers who use the component so I think we can stick with this approach. Thanks for the explanation @J-Son89.

One thing I think we could improve is documentation by adding docstrings. So that it will be easy to figure out what needs to be passed down to the component. Ref: https://guide.clojure.style/#documentation